### DISCUSSION TITLE
Chunk size out of benchmark !!

### DISCUSSION THREAD

**Discussion by iznotek** [vueform/vueform#316] [Questions]
Hye,
I have a nuxt3 app ongoing and i realize on the 'nuxt analyze' that vueform is taking way more size than expected and related in the benchmark section of the docs. I use the officlal vueform nuxt module has recommanded with full bundle default theme in vueform.config.ts. Any idea, step of optimization missed in vite or else ? Thanks for this amazing library !!

```
@vueform/vueform/dist/index.mjs of 504KB Rendered (GZIP 127KB)
@vueform/vueform/dist/vueform.mjs of 737KB Rendered (GZIP 175KB)
```

<img width="1659" alt="Capture d’écran 2024-09-01 à 07 58 11" src="https://github.com/user-attachments/assets/15e85db2-b15d-441a-9b4c-ed65d21fcf64">
<img width="1668" alt="Capture d’écran 2024-09-01 à 07 58 38" src="https://github.com/user-attachments/assets/8fe01c28-105b-4414-a258-2fa5cb65ed24">

**Comment by adamberecz** [comment]
Hey—there's a way to tree-shake things:
https://vueform.com/docs/tree-shaking

You might need to install it [manually](https://vueform.com/docs/installation#manual-installation) without using the Nuxt module by eg. using Vite installation steps and creating a client-only [plugin](https://nuxt.com/docs/guide/directory-structure/plugins) in Nuxt.

**Reply by iznotek** [reply]
thanks, i know for the tree-shake way... my point is how the whole lib take more than 1,2Mb ? the benchmark size in the tree-shaking section propose a 100kb for the whole lib ? any production regards on that ? thanks

**Reply by adamberecz** [reply]
Those are Gzipped sizes and the `Default components + all services` are about 100kb. If you include the whole lib that should be around 300-350kb (a few elements were added since the last benchmark).
